packages/store/src/storagecoder/StorageCoder.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { Slice, SliceLib } from "../Slice.sol";
+import { SchemaType } from "../Types.sol";
+
+library StorageCoder {
+  /**
+   * @dev Copies the array to the location of `packedSlice`,
+   * tightly packing it using the given size per element (in bytes)
+   */
+  function encodeTo(
+    bytes32[] memory array,
+    Slice packedSlice,
+    uint256 elementSize,
+    bool leftAligned // TODO this function is currently not used externally
+  ) private pure {
+    uint256 arrayLength = array.length;
+    uint256 packedPtr = packedSlice.ptr();
```

Comment:
https://github.com/latticexyz/mud/pull/387#discussion_r1106473625
I've thought about this actually, but since this function is temporary I'd rather put off optimizing it. Marked it with a clearer TODO